src/test/scala/com.snowplowanalytics.weather/providers/darksky/DarkSkyServerSpec.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2015-2018 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.weather
+package providers.darksky
+
+// Java
+import java.time.{Instant, ZoneOffset, ZonedDateTime}
+
+// Scala
+import scala.concurrent.duration._
+
+// cats
+import cats.effect.IO
+
+// tests
+import org.scalacheck.Gen
+import org.scalacheck.Prop.forAll
+import org.specs2.concurrent.ExecutionEnv
+import org.specs2.specification.ExecutionEnvironment
+import org.specs2.{ScalaCheck, Specification}
+
+// This library
+import com.snowplowanalytics.weather.Errors.AuthorizationError
+import com.snowplowanalytics.weather.providers.TestData
+
+/**
+ * Test case classes extraction from real server responses
+ */
+class DarkSkyServerSpec extends Specification with ScalaCheck with ExecutionEnvironment {
+  def is(implicit ee: ExecutionEnv) = skipAllIf(key.isEmpty) ^ s2"""
+
+    Test server responses (it can take several minutes)
+
+      big cities - forecast                 $e1
+      random cities - forecast              $e2
+      big cities - history                  $e3
+      random cities - history               $e4
+      sane error message for unauthorized   $e5
+      works with https                      $e6
+  """
+
+  private val key = sys.env.get("DARKSKY_KEY")
```

Comment:
are we fine running the tests with a free api key or will we need a paying one?

Reply:
1000 free calls per day, every following request requires added credit card and costs $0.0001. 

Current test suite uses 40 calls per run, so that gives us 12 free CI passes (two runs per pass) a day.